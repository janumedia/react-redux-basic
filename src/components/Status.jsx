import React from 'react';
import {connect} from 'react-redux';

import './Status.scss';

const mapStateToProps = state => state;

const Status = (props) => {
    const className = `Status ${props.error ? 'error' : props.pending ? 'pending' : ''}`;
    return (
        <p className={className}>{
            props.pending ? 'please wait...!' : (props.error ? props.error : '')
        }
        </p>
    )
}

export default connect(mapStateToProps)(Status);