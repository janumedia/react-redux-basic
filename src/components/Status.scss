.Status {
    padding: 14px 6px;
    transition:  color 0.5s background-color 1s;
    display: none;
    text-align: center;
    &.pending {
        display: block;
    }
    &.error {
        color: #fff;
        background-color: red;
        display: block;
    }
}