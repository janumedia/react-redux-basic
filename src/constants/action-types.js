const asyncActionTypes = type => ({
    PENDING: `${type}_PENDING`,
    SUCCESS: `${type}_SUCCESS`,
    FAILED: `${type}_FAILED`
});

export const GET_USERS = asyncActionTypes('GET_USERS');
export const ADD_USER = asyncActionTypes('ADD_USER');
export const DELETE_USER = 'DELETE_USER';