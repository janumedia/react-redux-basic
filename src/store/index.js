import { createStore, applyMiddleware } from 'redux';

import userReducer from '../reducers/userReducer';
import userMiddleWare from '../middleware/userMiddleWare';

export default createStore (userReducer, applyMiddleware(userMiddleWare));