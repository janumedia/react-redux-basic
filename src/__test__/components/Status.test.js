import React from 'react';
import renderer from 'react-test-renderer';

import store from '../../store';
import { GET_USERS, ADD_USER, DELETE_USER } from '../../constants/action-types';
import Status from '../../components/Status';

describe('Status component', () => {
    
    it('should display correct status', () => {
        const c = renderer.create(
            <Status store={store}/>
        );
        
        let tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: GET_USERS.PENDING});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: GET_USERS.FAILED, payload:"Error, get users failed!"});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: GET_USERS.SUCCESS, payload:[]});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: ADD_USER.PENDING});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: ADD_USER.FAILED, payload:"Error, add user failed!"});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: ADD_USER.SUCCESS, payload:{}});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();

        store.dispatch({type: DELETE_USER.FAILED, payload:"Error, delete user failed!"});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    
        store.dispatch({type: DELETE_USER.SUCCESS, payload:{}});
        tree = c.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});