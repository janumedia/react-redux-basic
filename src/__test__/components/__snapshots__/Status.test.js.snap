// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component should display correct status 1`] = `
<p
  className="Status "
>
  
</p>
`;

exports[`Status component should display correct status 2`] = `
<p
  className="Status pending"
>
  please wait...!
</p>
`;

exports[`Status component should display correct status 3`] = `
<p
  className="Status error"
>
  Error, get users failed!
</p>
`;

exports[`Status component should display correct status 4`] = `
<p
  className="Status "
>
  
</p>
`;

exports[`Status component should display correct status 5`] = `
<p
  className="Status pending"
>
  please wait...!
</p>
`;

exports[`Status component should display correct status 6`] = `
<p
  className="Status error"
>
  Error, add user failed!
</p>
`;

exports[`Status component should display correct status 7`] = `
<p
  className="Status "
>
  
</p>
`;
